### PR TITLE
Discourage using unrealistic catalog inputs in tests

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: 

## Summary

Since before the initial open source release of DocC, rdar://61945486 has tracked updating the tests to not use the "TestBundle" which contains a lot of inaccurate and unrealistic handcrafted data. Unfortunately, use of these test inputs have increased since then.

This PR renames those "TestBundle" to "DoNotUseInNewTests" to discourage using it in new tests and updates _many_ of the tests that only needed some content to created dedicated minimal test inputs to reduce the the use of these unrealistic test inputs. Lastly, it simplifies the loading of inputs in many tests.

>[!Note]
> This is a test-only PR.

## Dependencies

None.

## Testing

Nothing in particular. This isn't a user-facing change.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] ~Added~ Updated tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] ~Updated documentation if necessary~ Not applicable

### CODE REVIEW

File: Tests/SwiftDocCTests/Benchmark/ExternalTopicsHashTests.swift [hunk 1]
```diff
@@ -39,7 +39,7 @@ class ExternalTopicsGraphHashTests: XCTestCase {
     
     func testNoMetricAddedIfNoExternalTopicsAreResolved() throws {
         // Load bundle without using external resolvers
-        let (_, context) = try testBundleAndContext(named: "TestBundle")
+        let (_, context) = try testBundleAndContext(named: "DoNotUseInNewTests")
```
Comment (franklinsch) [inline Tests/SwiftDocCTests/Benchmark/ExternalTopicsHashTests.swift:42]: nit: Any thoughts on something more explicit like `LegacyBundle_DoNotUseInNewTests` to make it clearer that this is a test bundle?

**Review by franklinsch — APPROVED** [review]
Thanks!

File: Tests/SwiftDocCTests/Benchmark/ExternalTopicsHashTests.swift (hunk 1 above)
Comment (d-ronnqvist) [reply Tests/SwiftDocCTests/Benchmark/ExternalTopicsHashTests.swift:42]: I like it.

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Tests/SwiftDocCTests/Benchmark/ExternalTopicsHashTests.swift (hunk 1 above)
Comment (d-ronnqvist) [reply Tests/SwiftDocCTests/Benchmark/ExternalTopicsHashTests.swift]: Updated in https://github.com/swiftlang/swift-docc/pull/1106/commits/86ce80f96332ae40523d60f9482ad7db1834a6b7

**Pull request merged**